twoliter/src/project.rs

```diff
@@ -90,17 +99,39 @@ impl Project {
     pub(crate) fn project_dir(&self) -> PathBuf {
         self.project_dir.clone()
     }
+
+    pub(crate) fn sdk_name(&self) -> Option<&ImageName> {
+        self.sdk.as_ref()
+    }
+
+    pub(crate) fn sdk(&self, arch: &str) -> Option<ImageArchUri> {
+        self.sdk_name().map(|s| s.uri("sdk", arch))
+    }
+
+    pub(crate) fn toolchain(&self, arch: &str) -> Option<ImageArchUri> {
+        self.sdk_name().map(|s| s.uri("toolchain", arch))
+    }
 }
 
+/// A base name for an image that can be suffixed using a naming convention. For example,
+/// `registry=public.ecr.aws/bottlerocket`, `name=bottlerocket`, `version=v0.50.0` can be suffixed
+/// via naming convention to produce:
+/// - `registry=public.ecr.aws/bottlerocket/bottlerocket-sdk-x86_64:v0.50.0`
+/// - `registry=public.ecr.aws/bottlerocket/bottlerocket-toolchain-aarch64:v0.50.0`
```

Comment:
I'd like to get rid of the "magic" implicit suffixing if we can since ultimately that constrains the end user in kind of an arbitrary way.

What do you think about making the name a template like `bottlerocket-sdk-{{ arch }}`? That's consistent with similar patterns elsewhere in the project.

I would also recommend making the toolchain image its own explicit field, e.g. `toolchain_name` . But toolchain (and kmod kits) are just inherently problematic so we'll need to find a different way to deal with them at some point.

Reply:
I thought about this and didn't like the suffixing when I encountered it. I'm fine with changing to the non-magical version. It's not great to have a field in the schema that needs to be removed at some point. Any idea of a way to avoid having a schema key that is effectively temporary?

Reply:
It feels like the correct fix is to have the SDK always build a "native" toolchain, possibly in addition to the "cross" toolchain, so that the SDK itself always has the archive and we don't need to keep a second container in sync. We do something like this for Rust already, for different reasons.

I've opened https://github.com/bottlerocket-os/bottlerocket-sdk/issues/137 for that.

> Any idea of a way to avoid having a schema key that is effectively temporary?

Not really. We could have a fallback path, like "if the toolchain key isn't set, assume the SDK has the toolchain and use it instead".